tools/frr.in

```diff
@@ -18,6 +18,7 @@ D_PATH="@CFG_SBIN@" # /usr/lib/frr
 C_PATH="@CFG_SYSCONF@" # /etc/frr
 V_PATH="@CFG_STATE@" # /var/run/frr
 VTYSH="@vtysh_bin@" # /usr/bin/vtysh
+B_PATH="$(dirname $VTYSH)"
```

Comment:
Yeah I think this is not a great approach.  We should be setting the appropriate CFG_VTYSH_BIN(or other appropriately named thingie ) in configure.ac to load this variable.  Since vtysh can be placed in a directory besides /usr/bin( and it may not even be in the path, right? )

Reply:
`vtysh` binary is put under `bindir` (`$bindir/vtysh` in configure.ac (`vtysh_bin`)). I'm not sure how it should be (the correct way), to be honest. Expose `B_PATH="@CFG_BIN@"` or so?

Reply:
I'm pretty sure bindir is a configure option to specify on the configure cli, right?  Then it may be something non-standard

Reply:
The problem is that `/usr/lib/frr` is passed (sbindir, not bindir) to `frr-reload.py` as `--bindir` (which looks for where `vtysh` is installed). So, the fix is to pass the real bindir to `frr-reload.py`. `B_PATH="@bindir@"` might work too if this is fine?

Reply:
yes.... We should be able to grab the correct bin directory from the configure script run somewhere.  That is all the point I was making.  Especially if the end developer is compiling with a very funky setup( and it's totally possible )

Reply:
I'll do it to take it from configure.ac.

Reply:
Fixed.